packages/ws-server/src/webSocketServer.ts

```diff
@@ -87,6 +91,51 @@ const methodsCounterByIp = generateMethodsCounterById(register, {
   labelNames: WS_CONSTANTS.methodsCounterByIp.labelNames,
 });
 
+const updateResourceUtilizationMetrics: UpdateFunction = (): void => {
+  const cpuUsage = process.cpuUsage();
+  const totalCpuTime = cpuUsage.user + cpuUsage.system;
+  const cpuUsagePercentage = (totalCpuTime / os.cpus().length) * 100;
+  cpuUsageGauge.set(cpuUsagePercentage);
+
+  const memoryUsage = process.memoryUsage().rss;
+  memoryUsageGauge.set(memoryUsage);
+
+  const networkUsage =
+    os.networkInterfaces().eth0?.reduce((acc, iface) => acc + (iface as any).tx_bytes + (iface as any).rx_bytes, 0) ??
+    0;
+  networkUsageGauge.set(networkUsage);
+};
+
+const cpuUsageGauge = generateGauge(
+  register,
+  {
+    name: WS_CONSTANTS.cpuUsageGauge.name,
+    help: WS_CONSTANTS.cpuUsageGauge.help,
+    labelNames: WS_CONSTANTS.cpuUsageGauge.labelNames,
+  },
+  updateResourceUtilizationMetrics,
```

Comment:
all 3 gauges are using the same method. that would be inneficient since it would be called 3 times when it only needs to be done once, the specific for each gauge.

also, the set of the value is something like the provided example:

using the `this` keyword, setting the appropiate `labels` and then `set` the gauge value.

```      
    this.labels({ accountId: clientMain.operatorAccountId?.toString() }).set(accountBalance);
```